src/screencast/screencast.c

```diff
@@ -19,20 +19,30 @@ static const char object_path[] = "/org/freedesktop/portal/desktop";
 static const char interface_name[] = "org.freedesktop.impl.portal.ScreenCast";
 
 void exec_with_shell(char *command) {
-	pid_t pid = fork();
-	if (pid < 0) {
+	pid_t pid1, pid2;
+	int status;
+
+	if ((pid1 = fork())) {
+		waitpid(pid1, &status, 0);
+	} else if (pid1 == 0) {
+		if ((pid2 = fork())) {
+			exit(0);
```

Comment:
We should use `_exit` here instead, or else will end up hitting issues with finalizers. Might be a good idea to replace the one in the `execvp` error path below as well.